### ISSUE TITLE
node: pulumiLabels uniqueness error for a Cloud Run service in version 7.12.0

### ISSUE THREAD

**Issue by theneva** [pulumi/pulumi-gcp#1794]
### What happened?

Hi! :wave:

I'm upgrading from v6 to v7, and of course went for the latest release (`7.12.0`).

It really doesn't like my Cloud Run service:

<img width="1228" alt="image" src="https://github.com/pulumi/pulumi-gcp/assets/1404650/f5d833f8-911e-4c46-9a51-2c4ba574d89e">

Here's the error in text form, for copy-paste purposes:

```
unmarshaling urn:pulumi:prod::infra::folio:shared-infra$folio:turborepo-cache-service$gcp:cloudrun/service:Service::turborepo-cache-europe-west1's instance state: internal: Pulumi property 'terraformLabels' mapped non-uniquely to Terraform attribute 'terraform_labels' (duplicates Pulumi key 'pulumiLabels')
```

I've narrowed the problem down to `7.12.0`: I was able to upgrade by pinning `@pulumi/gcp` to `7.11.2`, and I can repeatedly do both `pulumi refresh` and `pulumi up` on `7.11.2`. Both commands fail on `7.12.0`.

### Example

I haven't spent any time on a minimal repro, but here's my Cloud Run Service resource – it's a Turborepo (https://turbo.build/repo/docs) server.

I'm omitting the other resources, as I don't see how they would affect the outcome. Please let me know if you need to see anything else!

Note that I'm not declaring any labels on any part of the service config.

```ts
const cloudrunService = new gcp.cloudrun.Service(
  name,
  {
    location,
    name,
    metadata: {
      annotations: {
        'run.googleapis.com/client-name': 'cloud-console',
        'run.googleapis.com/ingress': 'all',
        'autoscaling.knative.dev/maxScale': '2',
      },
    },
    template: {
      spec: {
        containerConcurrency: 420,
        timeoutSeconds: 10,
        containers: [
          {
            image,
            envs: [
              {
                name: 'TURBO_TOKEN',
                valueFrom: {
                  secretKeyRef: {
                    name: secret.secretId,
                    key: '1',
                  },
                },
              },
              { name: 'STORAGE_PROVIDER', value: 'google-cloud-storage' },
              { name: 'STORAGE_PATH', value: bucket.name },
              { name: 'NODE_ENV', value: 'production' },
              { name: 'LOG_LEVEL', value: 'debug' },
            ],
          },
        ],
        serviceAccountName: serviceAccount.email,
      },
    },
  },
  {
    parent,
    dependsOn: [turborepoTokenSecretAccessorBinding, secretVersion],
  },
);
```

I can't really see it mattering, but just in case…

The parent resource has these options:

```ts
{
  parent,
  aliases: [{ type: 'folio:tapico-service' }]
}
```

Its grandparent resource has these options:

```ts
{
  protect: true,
  providers: [gcpProvider]
}
```

`gcpProvider` is defined like this – note that I'm not declaring any labels of any kind:

```ts
const config = new Config();
const gcpConfig = types.gcpConfigRt.check(config.requireObject('gcpConfig'));

const environment = getStack();

const gcpProviderConfig = {
  project: gcpConfig.project,
  region: gcpConfig.region,
  zone: gcpConfig.zone,
};

const gcpProvider = new gcp.Provider('gcp_provider', gcpProviderConfig, {
  protect: true,
});
```

### Output of `pulumi about`

```
$ pulumi about

CLI
Version      3.108.1
Go Version   go1.22.0
Go Compiler  gc

Plugins
NAME    VERSION
nodejs  unknown

Host
OS       darwin
Version  14.1.1
Arch     arm64

This project is written in nodejs: executable='/Users/theneva/.nvm/versions/node/v20.11.1/bin/node' version='v20.11.1'

Backend
Name           Martins-MacBook-Pro.local
URL            gs://<redacted>
User           theneva
Organizations
Token type     personal
```

### Additional context

I'm using a Google Cloud Storage bucket as my backend, if that matters. I also have the env var `PULUMI_SKIP_CHECKPOINTS=true` set to speed things up, since checkpoints are _slow_ on GCS backends.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by iwahbe** [comment]
Hi @theneva. Thanks for letting us know.

**Comment by theneva** [comment]
> We are experiencing the same issue after upgrading to "@pulumi/pulumi": "^3.108.1" and "@pulumi/gcp": "^7.12.0".

I'm sure you figured it out, but just in case: Pinning @pulumi/gcp  to "7.11.2" instead of "^7.12.0" avoids the issue.

I don't think the @pulumi/pulumi version is part of the problem - but please speak up if it is!

**Comment by antebandov** [comment]
> > We are experiencing the same issue after upgrading to "@pulumi/pulumi": "^3.108.1" and "@pulumi/gcp": "^7.12.0".
> 
> I'm sure you figured it out, but just in case: Pinning @pulumi/gcp to "7.11.2" instead of "^7.12.0" avoids the issue.
> 
> I don't think the @pulumi/pulumi version is part of the problem - but please speak up if it is!

Thank you for the quick reply. Pinning @pulumi/gcp to "7.11.2" solved the issue.

**Comment by theneva** [comment]
@iwahbe After upgrading all my projects and stacks to `7.11.2`, I've noticed that Cloud Run service resources across the board seem to dislike the new `pulumiLabels` property: I consistently get the diff you can see in the image on every `pulumi up`, but only for Cloud Run resources.

Maybe that points to specifically the `gcp.cloudrun.Service` resource missing some kind of handling of the new label properties?

@antebandov do you see the issue on any resource types other than Cloud Run Services? Are you also using Node.js?

![image](https://github.com/pulumi/pulumi-gcp/assets/1404650/e61387e3-d5cf-4264-9825-31cf80e6faf5)

**Comment by antebandov** [comment]
@theneva I‘m also using Node.js but haven’t had any issues regarding the pulumiLabels after downgrading to 7.11.2.

**Comment by tenhaus** [comment]
Seeing this too with Service and DomainMapping

**Comment by gchristov** [comment]
> Thank you for the quick reply. Pinning @pulumi/gcp to "7.11.2" solved the issue.

I posted in the other thread too as it was the first one I found but this also resolved my issue.

What is the best way to avoid these transitive dependency issues going forward? Do y'all pin your GCP versions for all your resources? I've got around 100 of them so that doesn't seem great 😅

**Comment by antdking** [comment]
It looks like this issue has been accidentally marked as resolved; can it be re-opened until the fixed version of tfbridge is released and used for this provider please

**Comment by theneva** [comment]
> It looks like this issue has been accidentally marked as resolved; can it be re-opened until the fixed version of tfbridge is released and used for this provider please

For what it's worth, the fix was released in 7.14.1 earlier today: https://github.com/pulumi/pulumi-gcp/releases/tag/v7.14.1
